src/sharing/components/Recipient.jsx

```diff
@@ -35,34 +36,55 @@ export const UserAvatar = ({ name, url }) => (
   </div>
 )
 
-const Recipient = ({ t, contact, status, documentType, onUnshare }) => {
-  const name = getPrimaryEmail(contact)
-  const url = getPrimaryCozy(contact)
-  return (
-    <div className={styles['pho-recipient']}>
-      <Avatar name={name} />
-      <Identity name={name} url={url} />
-      {status && (
+class Recipient extends Component {
+  state = {
+    revoking: false
+  }
+
+  onUnshare = () => {
+    this.setState({ revoking: true })
+    this.props
+      .onUnshare(this.props.contact)
+      .then(() => this.setState({ revoking: false }))
+  }
+
+  render() {
+    const { t, contact, status, type, documentType } = this.props
+    const { revoking } = this.state
+    const name = getPrimaryEmail(contact)
+    const url = getPrimaryCozy(contact)
+    return (
+      <div className={styles['pho-recipient']}>
+        <Avatar name={name} />
+        <Identity name={name} url={url} />
         <div className={styles['pho-recipient-status']}>
```

Comment:
It will be clearer to refactor this underlying component (everything under `<div className={styles['pho-recipient-status']}>`) into its own component definition with a HOC to display the spinner or the component.

Reply:
OK for refactoring into a Status component, but I can't find a cool API for a HOC like you suggest. Any idea?

Reply:
Maybe with something like that: https://codesandbox.io/s/zr49prv66p

```jsx
const withSpinner = (isLoading, BaseComponent) => props =>
  isLoading(props) ? <Spinner /> : <BaseComponent {...props} />;
```